akvo-top-asn/akvo-top-asn

```diff
@@ -163,24 +163,44 @@ def get_connection(config):
 
 
 def query_clickhouse(time_range, direction):
-    query = f"""
-        WITH
-        source AS (SELECT * FROM flows_5m0s SETTINGS asterisk_include_alias_columns = 1),
-        rows AS (SELECT SrcAS FROM source WHERE TimeReceived BETWEEN toDateTime('{time_range[0]}', 'UTC') AND toDateTime('{time_range[1]}', 'UTC') AND (InIfBoundary = 'external') GROUP BY SrcAS ORDER BY SUM(Bytes) DESC LIMIT 50)
```

Comment:
@rebortg I dont really understand why we need those two queries?

wouldn't `{direction}IfBoundary = 'exertnal'` in this line also get the job done?

Reply:
The original query was broken in several ways:
- not all interface columns where adopted to the `direction` setting
- it has only evaluated the traffic of the `SrcAS`

Splitting the query seems to be the best way to make it easier to understand.